File: python/exercício_17.py
print()
print("=-"*60)
print()
print("---------------Saque no caixa el------------------")
print()
print("=-"*60)
print()

valor=int(input("Digite o valor que deseja sacar: "))
total = valor
celula=100
totalcelula=0
while True:
    if total >= celula:
        total -= celula
        totalcelula += 1
    else:
        if totalcelula > 0:
            print(f'Total de {totalcelula} cédulas de R${celula}')
        if celula == 100:
            celula = 50 
        elif celula == 50:
            celula = 20
        elif celula == 20:
            celula = 5
        elif celula == 5:
            celula = 2 
        elif celula == 2:
            celula = 1
        elif celula == 1:
            celula = 1
        totalcelula = 0
        if total == 0:
            break 

File: java/exercício_17.java
import java.util.Scanner;

public class exercício_17 {
    public static void main(String[] args) {
        try (Scanner b = new Scanner(System.in)) {
            System.out.println("Digite o valor que deseja:");
            int valor = b.nextInt();
            int total = valor;
            int celula = 100;
            int totalcelula = 0;

            while (true)
                if (total >= celula) {
                    total -= celula;
                    totalcelula++;
                } else {
                    if (totalcelula > 0) {
                        System.out.println("Total de " + totalcelula + "cédulas de R$" + celula);
                    }
                    if (celula == 100) {
                        celula = 50;
                    } else if (celula == 50) {
                        celula = 20;
                    } else if (celula == 20) {
                        celula = 5;
                    } else if (celula == 5) {
                        celula = 2;
                    } else if (celula == 2) {
                        celula = 1;
                    } else if (celula == 1) {
                        celula = 1;
                    }
                    totalcelula = 0;
                    if (total == 0) {
                        break;
                    }
                }
        }

    }
}